lib/modules/datasource/go/index.ts

```diff
@@ -59,8 +59,11 @@ export class GoDatasource extends Datasource {
       return null;
     }
 
-    // ignore v0.0.0- pseudo versions that are used Go Modules - look up default branch instead
-    const tag = value && !value.startsWith('v0.0.0-2') ? value : undefined;
+    // ignore vX.Y.Z-(0.)? pseudo versions that are used Go Modules - look up default branch instead
+    const tag =
+      value && !/^v\d+\.\d+\.\d+-0?\.?\d{14}-([a-f0-9]{12})/.test(value)
```

Comment:
```suggestion
      value && !/^v\d+\.\d+\.\d+-(?:0\.)?\d{14}-([a-f0-9]{12})/.test(value)
```

still doesn't support `1.2.3-pre.0.yyyymmdd` like pseudo versions.

the regex should be centrally defined inside go datasource and be reused by the manager.